### ISSUE TITLE
readiness probe failing for webservice pod 

### ISSUE THREAD

**Issue by auto-store** [NetApp/devops-at-scale#11]
all pods come up successfully but webservice pod fails,  readiness probe reports `statuscode: 500 `

I have increased liveness timeouts in the helm chart:  `devops-at-scale/charts/webservice/templates/webservice-deployment.yaml`

output of kubectl describe "webservice pod" :

```
Name:               devops-at-scale-webservice-768dc8859f-7q7p5
Namespace:          default
Priority:           0
PriorityClassName:  <none>
Node:               k8s063/192.168.100.164
Start Time:         Tue, 30 Jul 2019 10:06:14 +0100
Labels:             app=devops-at-scale-webservice
                    pod-template-hash=768dc8859f
Annotations:        <none>
Status:             Running
IP:                 10.244.1.21
Controlled By:      ReplicaSet/devops-at-scale-webservice-768dc8859f
Containers:
  devops-at-scale-webservice:
    Container ID:   docker://7ac998f245dd9f15fc87d443c18e4dbb6c3c6d4aef2622bc034f984f4c280d20
    Image:          archana51189/devops-at-scale:1.2
    Image ID:       docker-pullable://archana51189/devops-at-scale@sha256:4a46c23dbf110c5ca6529f2812fd8007243acb560438fe1eb97f92702f105302
    Ports:          80/TCP, 6000/TCP
    Host Ports:     0/TCP, 0/TCP
    State:          Running
      Started:      Tue, 30 Jul 2019 10:06:16 +0100
    Ready:          False
    Restart Count:  0
    Liveness:       http-get http://:http/ delay=600s timeout=1s period=40s #success=1 #failure=600
    Readiness:      http-get http://:http/ delay=600s timeout=1s period=40s #success=1 #failure=600
    Environment:
      SCM_TYPE:               bitbucket
      SCM_SERVICE_NAME:       devops-at-scale-bitbucket
      SCM_PVC_NAME:           devops-at-scale-bitbucket-pvc
      JENKINS_SERVICE_NAME:   devops-at-scale-jenkins
      JENKINS_PVC_NAME:       devops-at-scale-jenkins-pvc
      DATABASE_SERVICE_NAME:  devops-at-scale-couchdb
      DATABASE_PVC_NAME:      devops-at-scale-couchdb-pvc
      REGISTRY_SERVICE_NAME:  devops-at-scale-artifactory
      REGISTRY_PVC_NAME:      devops-at-scale-artifactory-pvc
      REGISTRY_TYPE:          artifactory
      WEB_SERVICE_NAME:       devops-at-scale-webservice
      WEB_PVC_NAME:           devops-at-scale-webservice-pvc
      SERVICE_TYPE:           NodePort
      KUBE_NAMESPACE:         default
      STORAGE_CLASS:          basic
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from devops-at-scale-webservice-token-wq756 (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  devops-at-scale-webservice-token-wq756:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  devops-at-scale-webservice-token-wq756
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason     Age                  From               Message
  ----     ------     ----                 ----               -------
  Normal   Scheduled  25m                  default-scheduler  Successfully assigned default/devops-at-scale-webservice-768dc8859f-7q7p5 to k8s063
  Normal   Pulling    25m                  kubelet, k8s063    Pulling image "archana51189/devops-at-scale:1.2"
  Normal   Pulled     25m                  kubelet, k8s063    Successfully pulled image "archana51189/devops-at-scale:1.2"
  Normal   Created    25m                  kubelet, k8s063    Created container devops-at-scale-webservice
  Normal   Started    25m                  kubelet, k8s063    Started container devops-at-scale-webservice
  Warning  Unhealthy  5m6s (x16 over 15m)  kubelet, k8s063    Liveness probe failed: HTTP probe failed with statuscode: 500
  Warning  Unhealthy  7s (x23 over 14m)    kubelet, k8s063    Readiness probe failed: HTTP probe failed with statuscode: 500
```
attempted in separate isolated clusters with no success.

**Comment by engineerakki** [comment]
Hello @auto-store ,

Can you please let us know following things so that we can reproduce the issue.

K8s distribution:  Native OnPrem K8s/NKS etc.  
K8s version:   (eg:- 1.11/1.12/1.13/1.14)
Trident version:

**Comment by auto-store** [comment]
Hi @engineerakki 

K8s distribution: onprem - VMware VM's (centos) 
K8s version: 1.14.1
Trident version: 19.07

**Comment by forgosh** [comment]
> Hello @auto-store ,
> 
> Can you please let us know following things so that we can reproduce the issue.
> 
> K8s distribution: Native OnPrem K8s/NKS etc.
> K8s version: (eg:- 1.11/1.12/1.13/1.14)
> Trident version:

k8s on prem - VMware VMs (CentOS 7)
k8s version - 1.15.1
Trident version 19.07

**Comment by vv-io** [comment]
@lonico gentle bump as am too trying to get this going, so a fix would be very welcome.

k8s on prem - VMware VMs (CentOS 7)
k8s version - 1.15.1
Trident version 19.07

**Comment by lonico** [comment]
This is still under code review, but if you are brave enough, I have a new container image that solves the CSI issue.  I plan to do more validations tomorrow.

You can use it by editing:

devops-at-scale/charts/webservice/values.yaml
and changing the image: line to:

image: lonico/devops-at-scale:1.2.0.2
Then, assuming you are in the devops-at-scale directory

helm del --purge devops-at-scale
helm install --name devops-at-scale .

**Comment by forgosh** [comment]
Will deploy later this morning....

Seth Forgosh
Solution Architect - Open Ecosystem, Virtualization & Flash

________________________________

**Comment by forgosh** [comment]
Worked like a champ.. now to start using it

**Comment by lonico** [comment]
fixed by #13
